src/utils/hmac.js

```diff
@@ -17,13 +19,32 @@ function createHmacHelper (hmacKey) {
     return hmac(iprConditionKey, jsonString)
   }
 
-  function getIprReceiverId () {
+  function getReceiverId () {
     return hmac(hmacKey, IPR_RECEIVER_ID_STRING).slice(0, 8)
   }
 
+  function getSspToken () {
+    return crypto.randomBytes(16)
+  }
+
+  function getSspSharedSecret (token) {
+    const generator = hmac(hmacKey, SSP_GENERATION_STRING)
+    return hmac(generator, token).slice(0, 16)
+  }
+
+  function hmacJsonForSspCondition (obj, sharedSecret) {
+    const sspConditionKey = hmac(sharedSecret, SSP_CONDITION_STRING)
+    const jsonString = stringify(obj)
+    const hmacDigest = hmac(sspConditionKey, jsonString)
+    return hmacDigest
+  }
+
   return {
     hmacJsonForIprCondition,
-    getIprReceiverId
```

Comment:
How come IPR receiver has been changed to just receiver in the code? Wouldn't it make sense to be more specific now that we have two protocols being used in here?